### ISSUE TITLE
Problem connecting in Https to qBittorrent

### ISSUE THREAD

**Issue by Itay1787** [tympanix/Electorrent#157]
**Client:**
- [ ] µTorrent
- [x] qBittorrent
- [ ] Transmission
- [ ] rTorrent
- [ ] Deluge

**Client Version:**
qBittorrent v4.1.6
**Operating System:**
qBittorrent in FreeNAS Electorrent in Windows 10
**Application Version:**
2.6.0
**Description:**
Hi I installed Electorrent in Win10 and qBittorrent v4.1.6 in FreeNAS
And I have a problem connecting to qbit in Electorrent at Https
At first I opened a cert on the CA I had already created for FreeNAS. I copied Key and Certificate to qbit and worked in the browser with a green lock and everything is OK
Then I tried to connect to Electorrent and he introduced me to the error:
`Invalid certificate chain The certificate could not be verified because the certificate chain is invalid. Consolidate your webserver TLS configuration`

So I created a new CA and a new Cert in FreeNAS (it's just a lot more convenient to do it there) and all the details are equal both in CA and in Cert

I put the Cert into the qbit and put the CA into the browser I have a green lock everything works
Then I tried to connect again in Electorrent and now and it gives me the option to click Trust and after that and I have another error:
`Certificate error The certificate is not useable with this server because the common name of the certificate does not match the hostname of the server`
I inserted in CN both in CA and in Cert the same IP I am trying to connect to now and it does not work I do not know what to do It looks like bug
Asking for help .... please
Thanks
Itay

**Comment by tympanix** [comment]
The first error you get is because the certificate you use does not contain the full chain to the root certificate. Generate and use `fullchain.pem` instead. Problem is similar to [using Let's Encrypt](https://github.com/tympanix/Electorrent/wiki#using-tls-with-lets-encrypt).

The second error you get is the common name. Usually you would put the domain in here, or just `localhost` if it was the same machine. Your browser might ignore this but the NodeJs net library is particularly picky about the common name. I have not personally tried with IP addresses in the common name, only with domain names. If you can give your FreeNAS a domain name on your LAN that should work out, else I would urge you to try and fiddle a bit more with the common name and IP address. If nothing works out I might have to rethink this scenario and change some code in Electorrent.

Thanks for reporting 👍

**Comment by tympanix** [comment]
@Itay1787 any news on this?

**Comment by Itay1787** [comment]
> 
> 
> @Itay1787 any news on this?

Yes, the only way I could get it to work properly was by opening DDNS and pointing the IP to my internal address. I created a letencrypt to DDNS and moved HTTPS through nginx reverse proxy then everything worked including browser

**Comment by tympanix** [comment]
Ok great, so it seems that nodejs will not accept certificates with IP addresses in the common name as suspected. I'll keep the issue open as to come back to this

**Comment by tympanix** [comment]
IPv4 addresses in common names is not valid for x509 identify verification.

Use SAN/subjectAltName instead. See RFC 5280
